### PR TITLE
fix(controller): rm sub-resources from cluster api enablement list

### LINKED ISSUE #465: Remove sub-resources from cluster api enablement list
**What would you like to be added**:
Remove the redundant API resources from the [API Resoruces](https://github.com/karmada-io/karmada/blob/c053b6c8b324fe48f70bd88c970074de9fd0df7f/pkg/apis/cluster/v1alpha1/types.go#L143).

E.g, the `namespaces/finalize`, `namespaces/status`, etc are not the real API resources that we are caring about.
```
apiVersion: cluster.karmada.io/v1alpha1
kind: Cluster
metadata:
  creationTimestamp: "2021-06-24T06:18:12Z"
  finalizers:
  - karmada.io/cluster-controller
  name: member1
  resourceVersion: "319"
  selfLink: /apis/cluster.karmada.io/v1alpha1/clusters/member1
  uid: a30fb780-6b20-4535-8a86-d5a64c415ff9
spec:
  apiEndpoint: https://172.18.0.5:6443
  secretRef:
    name: member1
    namespace: karmada-cluster
  syncMode: Push
status:
  apiEnablements:
  - groupVersion: v1
    resources:
    - kind: Namespace
      name: namespaces
    - kind: Namespace
      name: namespaces/finalize
    - kind: Namespace
      name: namespaces/status
    - kind: Node
      name: nodes
    - kind: NodeProxyOptions
      name: nodes/proxy
    - kind: Node
      name: nodes/status
    - kind: PersistentVolumeClaim
      name: persistentvolumeclaims
    - kind: PersistentVolumeClaim
      name: persistentvolumeclaims/status
    - kind: PersistentVolume
      name: persistentvolumes
    - kind: PersistentVolume
      name: persistentvolumes/status
    ...
```

**Why is this needed**:
Removing redundant info for clean and tidy.

In contrast, the `kubectl api-resources` ignored all the sub-resources:
```
# kubectl api-resources | grep -i namespaces
namespaces                        ns                                          false        Namespace
```

### PR DESCRIPTION
**Pull request by shinytang6**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #465

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/controllers/status/cluster_status_controller.go [hunk 1]
```diff
@@ -371,6 +372,10 @@ func getAPIEnablements(clusterClient *util.ClusterClient) ([]v1alpha1.APIEnablem
 	for _, list := range apiResourceList {
 		var apiResources []v1alpha1.APIResource
 		for _, resource := range list.APIResources {
+			if strings.Contains(resource.Name, "/") {
```
Comment (RainbowMango) [inline pkg/controllers/status/cluster_status_controller.go:375]: Can you help add a comment here?

Comment (shinytang6) [reply pkg/controllers/status/cluster_status_controller.go:376]: ok, done

Comment (RainbowMango) [reply pkg/controllers/status/cluster_status_controller.go:376]: Thanks.

**Pull request merged**